src/service.c

```diff
@@ -984,7 +985,72 @@ static void indicator_a11y_service_init (IndicatorA11yService *self)
     self->pPrivate->sUser = NULL;
     self->pPrivate->bReadingAccountsService = FALSE;
     self->pPrivate->sHighContrast = NULL;
+    self->pPrivate->bVirtualX = FALSE;
     GError *pError = NULL;
+
+    // Check if we are in a virtual environment
+    Display *pDisplay = XOpenDisplay (NULL);
+
+    if (!pDisplay)
+    {
+        g_error ("Panic: Failed to open X display");
+
+        return;
+    }
+
+    guint nScreen = DefaultScreen (pDisplay);
+    Window pWindow = RootWindow (pDisplay, nScreen);
+    XRRScreenResources *pResources = XRRGetScreenResources (pDisplay, pWindow);
+
+    if (!pResources)
+    {
+        g_error ("Panic: Failed to get screen resources");
+        XCloseDisplay (pDisplay);
+
+        return;
+    }
```

Comment:
We should limit this panic to desktop environments that use X11. Like: is WAYLAND_DISPLAY set, then ignore all of this. I know, a11y indicators is targetting MATE only for now, but we should imagine, a11y being for more than one desktop env and it might be a non-X11 env. Or: a11y should also run on MATE when it gets its port to Wayland some time in the future.

Long story cut short: Please only run xlib code when we are running in X11.

Reply:
I would like to have this clean and check only for a virtual environment here. It would prefer to check for Wayland in a separate PR after this is merged.

Reply:
Ok, noted. https://github.com/AyatanaIndicators/ayatana-indicator-a11y/issues/30#issuecomment-2228755277